ts/main.ts

```diff
@@ -21,14 +28,96 @@ $urlLinkInput.addEventListener('input', (event: Event) => {
 $entryForm.addEventListener('submit', (event: Event) => {
   event.preventDefault();
   const $formElements = $entryForm.elements as FormElements;
-  const $formObject = {
+  const $formObject: FormObject = {
     title: $formElements.title.value,
     url: $formElements.url.value,
     notes: $formElements.notes.value,
     entryID: data.nextEntryId,
   };
   data.nextEntryId++;
   data.entries.unshift($formObject);
+  const newEntry = renderEntry($formObject);
+  const $ul = document.querySelector('ul');
+  $ul?.prepend(newEntry);
+  viewSwap('entries');
+  if (data.entries.length > 0) {
+    toggleNoEntries();
+  }
   $entryForm.reset();
   $previewPhoto?.setAttribute('src', 'images/placeholder-image-square.jpg');
 });
+
+function renderEntry(entry: FormObject): HTMLLIElement {
+  const containingLi = document.createElement('li');
+  const containingRowDiv = document.createElement('div');
+  containingRowDiv.classList.add('row');
+  const imgColDiv = document.createElement('div');
+  imgColDiv.classList.add('column-full', 'column-half');
+  const textColDiv = document.createElement('div');
+  textColDiv.classList.add('column-full', 'column-half');
+  const titleRowDiv = document.createElement('div');
+  titleRowDiv.classList.add('row');
+  const titleColDiv = document.createElement('div');
+  titleColDiv.classList.add('column-full');
+  const parRowDiv = document.createElement('div');
+  parRowDiv.classList.add('row');
+  const parColDiv = document.createElement('div');
+  parColDiv.classList.add('column-full');
+  const image = document.createElement('img');
+  image.setAttribute('src', entry.url);
+  const title = document.createElement('h4');
+  title.textContent = entry.title;
+  const paragraph = document.createElement('p');
```

Comment:
Since these are storing DOM elements, I would put a `$` in front of the variable names